### PR TITLE
[ST] Test CruiseControl Resources/JVM options

### PR DESCRIPTION
**Pull request by michalxo**
Signed-off-by: Michal T <mtoth@redhat.com>

Added Test to check CruiseControl resources and JVM arguments.

### CODE REVIEW

**Comment by see-quick** [comment]
Why such a test case is needed? It seems to me that this can be verified on the `UNIT` level. Is there some hidden motivation for this, which I don't see? :)

**Review by scholzj — COMMENTED** [review]
Is this really worth separate test instead of adding it to some existing test? Also, I wonder what are the resources based on. They seem to be fairly high given our resource issues in STs.

File: systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java [hunk 1]
```diff
@@ -67,6 +70,38 @@ public class CruiseControlST extends AbstractST {
...
+            .endSpec()
+            .build());
+
+        String ccPodName = kubeClient().listPodsByPrefixInName(namespace, CruiseControlResources.deploymentName(clusterName)).get(0).getMetadata().getName();
+        assertResources(namespace, ccPodName, "cruise-control",
+                "1.5Gi", "1", "1Gi", "800m");
+        assertExpectedJavaOpts(namespace, ccPodName, "cruise-control",
+                "-Xmx1G", "-Xms256M", "-XX:+UseG1GC");
+    }
+
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java:104]: The test itself is good, but I think you can add it to another test, which is already written. My point is that you will wait for example 5 minutes for Kafka and CC will be up and then you just check that resources are correct - which is little bit small check for the extra time.

**Comment by scholzj** [comment]
> Why such a test case is needed? It seems to me that this can be verified on the UNIT level. Is there some hidden motivation for this, which I don't see? :)

The asserting of the Java options is not something what is really tested in unit tests since that is about actually running the operand. How critical it is to cover that with ST, that is another discussion I guess.

File: systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java (hunk 1 above)
Comment (michalxo) [reply systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java:104]: Yup, already added it into `testAutoCreationOfCruiseControlTopicsWithResources` (and changed name). 
I think it won't affect original test in any way. Agree?

File: systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java [hunk 2]
```diff
@@ -79,9 +81,28 @@ void testAutoCreationOfCruiseControlTopics(ExtensionContext extensionContext) {
...
                 .endKafka()
+                .editCruiseControl()
+                    .withResources(new ResourceRequirementsBuilder()
+                        .addToLimits("memory", new Quantity("256Mi"))
+                        .addToLimits("cpu", new Quantity("1"))
+                        .addToRequests("memory", new Quantity("256Mi"))
+                        .addToRequests("cpu", new Quantity("0.8"))
+                        .build())
+                    .withNewJvmOptions()
+                        .withXmx("512M")
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java:92]: you cannot fit Xmx512M into 256Mi. You probably cannot fit -Xms256M either.

File: systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java [hunk 3]
```diff
@@ -79,9 +81,28 @@ void testAutoCreationOfCruiseControlTopics(ExtensionContext extensionContext) {
                 .editKafka()
                     .addToConfig("auto.create.topics.enable", "false")
                 .endKafka()
+                .editCruiseControl()
+                    .withResources(new ResourceRequirementsBuilder()
+                        .addToLimits("memory", new Quantity("256Mi"))
+                        .addToLimits("cpu", new Quantity("1"))
+                        .addToRequests("memory", new Quantity("256Mi"))
+                        .addToRequests("cpu", new Quantity("0.8"))
+                        .build())
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java:90]: Maybe leaving out the CPU completely would be better -> lower chance of conflict or of running out of resources.

File: systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java (hunk 2 above)
Comment (michalxo) [reply systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java:92]: Aaah, I see now. Of course it's connected to container limits.

File: systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java (hunk 3 above)
Comment (michalxo) [reply systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java:90]: I will leave there 0.1 CPU for both, as we don't have such test for CC. Would you approve it with that? Else I would need to duplicate code from `assertResources()` and also possibly write new test as well for that CPU checks :-)

Comment (michalxo) [reply systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java:90]: Update: Will remove CPU, as CC is not able to start with 0.1 CPU.

File: systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java [hunk 4]
```diff
@@ -79,9 +84,27 @@ void testAutoCreationOfCruiseControlTopics(ExtensionContext extensionContext) {
...
+                        .withXx(Map.of("UseG1GC", "true"))
+                    .endJvmOptions()
+                .endCruiseControl()
             .endSpec()
             .build());
 
+        String ccPodName = kubeClient().listPodsByPrefixInName(namespace, CruiseControlResources.deploymentName(clusterName)).get(0).getMetadata().getName();
+        Container container = (Container) KubeClusterResource.kubeClient(namespace).getPod(namespace, ccPodName).getSpec().getContainers().stream().filter(c -> c.getName().equals("cruise-control")).findFirst().get();
+        assertThat(container.getResources().getLimits().get("memory"), CoreMatchers.is(new Quantity("300Mi")));
+        assertThat(container.getResources().getRequests().get("memory"), CoreMatchers.is(new Quantity("300Mi")));
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java:104]: `CoreMatchers.is()` -> `is()`

**Review by im-konge — APPROVED** [review]
LGTM, one nit :)

File: systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java [hunk 5]
```diff
@@ -79,9 +83,27 @@ void testAutoCreationOfCruiseControlTopics(ExtensionContext extensionContext) {
...
                 .endKafka()
+                .editCruiseControl()
+                    .withResources(new ResourceRequirementsBuilder()
+                        .addToLimits("memory", new Quantity("300Mi"))
+                        .addToRequests("memory", new Quantity("300Mi"))
+                        .build())
+                    .withNewJvmOptions()
+                        .withXmx("200M")
+                        .withXms("128M")
+                        .withXx(Map.of("UseG1GC", "true"))
```
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java:94]:
```suggestion
                        .withXx(Collections.singletonMap("UseG1GC", "true"))
```

**Review by see-quick — APPROVED** [review]
Looks good, just a nuance.

**Review by scholzj — APPROVED** [review]

**Pull request merged**